drivers/serial/uart_mcux_lpuart.c

```diff
@@ -1016,22 +1016,43 @@ static int mcux_lpuart_configure_basic(const struct device *dev, const struct ua
 	}
 #endif
 
-#if defined(FSL_FEATURE_LPUART_HAS_MODEM_SUPPORT) && \
-	FSL_FEATURE_LPUART_HAS_MODEM_SUPPORT
 	switch (cfg->flow_ctrl) {
 	case UART_CFG_FLOW_CTRL_NONE:
+#ifdef LPUART_HAS_MODEM
+		uart_config->enableTxCTS = false;
+		uart_config->enableRxRTS = false;
+#endif
+		break;
+
 	case UART_CFG_FLOW_CTRL_RS485:
+#ifdef LPUART_HAS_MODEM
 		uart_config->enableTxCTS = false;
 		uart_config->enableRxRTS = false;
 		break;
+#else
+		return -ENOTSUP;
+#endif
+
 	case UART_CFG_FLOW_CTRL_RTS_CTS:
+#ifdef LPUART_HAS_MODEM
 		uart_config->enableTxCTS = true;
 		uart_config->enableRxRTS = true;
 		break;
+#else
+		return -ENOTSUP;
+#endif
+
+	case UART_CFG_FLOW_CTRL_DTR_DSR:
+#if defined(FSL_FEATURE_LPUART_HAS_MCR) && FSL_FEATURE_LPUART_HAS_MCR
+		/* DTR/DSR enabled, no config needed */
+		break;
+#else
+		return -ENOTSUP;
+#endif
```

Comment:
instead of a bunch of cases with a bunch of #if #else, just check like

```
if (!IS_ENABLED(LPUART_HAS_MODEM) && cfg->flow_ctrl != UART_CONFIG_FLOW_CTRL_NONE) {
  return -ENOTSUP;
}
```

at the start of the function, similar with the MCR expression using another macro like suggested above

and consider making helper functions